crates/telio-model/src/features.rs

```diff
@@ -369,10 +369,13 @@ impl FeatureEndpointProvidersOptimization {
     }
 }
 
-/// Turns on connection resets upon VPN server change
+/// Feature config for firewall
 #[derive(Default, Clone, Copy, Debug, PartialEq, Eq, Deserialize)]
-#[serde(transparent)]
-pub struct FeatureBoringtunResetConns(pub bool);
+pub struct FeatureFirewall {
+    /// Turns on connection resets upon VPN server change
+    #[serde(default)]
+    pub boringtun_reset_conns: bool,
+}
```

Comment:
This is a breaking change - is this indented? Maybe it would be better to avoid it, and use something like `[serde(flatten)]` to keep it compatible.

Reply:
I think the breaking change is okay since it would go out with a new major version. If it makes more sense to avoid the breakage then let's go with that, but since the next release is 5.0, we don't have to worry much about breaking changes, especially not in feature and meshnet configs since apps will most likely use the `deserialize_*_config' functions. So as long as we can parse both the new and the old way of doing this, it should be fine

Reply:
Good point, although I still think that breaking compatibility when a fix is easy should never be done. So feel free to treat this as a nitpick and ignore it if you see it differently :)

Reply:
I highly agree that we shouldn't break compat if it's not necessary and I'm not really weighing in on this specific case, just putting it out there that if we need to break compat, now is a great time :smile:

Reply:
In this case, the breaking change is intentional, as this PR has been specifically created to include this change in major libtelio release. 

More firewall configuration options in the FeatureConfig are coming soon :tm:, therefore we wanted to group them into single group. 

But as you correctly identified - this is a breaking change, and this is the reason for separating this PR from actual firewall changes, as we would like to introduce changes with libtelio v5.0 release, which is already breaking change.